### PR TITLE
Refs #11593 -- Added testing to resuable apps intro

### PR DESCRIPTION
**Pull request by RaphaelKimmig**
Added a section about testing resuable apps to the advanced tutorial.

### CODE REVIEW

File: docs/intro/reusable-apps.txt [hunk 1]
```diff
@@ -288,6 +288,70 @@ working. We'll now fix this by installing our new ``django-polls`` package.
 
     pip uninstall django-polls
 
+Testing your app
+================
+Since we moved the ``polls`` directory out of the project,
+we don't have the project's ``settings.py`` and ``urls.py`` anymore.
+We need another way to configure our app in order to test it.
+
+#. First we create a ``tests`` module with an empty ``__init__.py`` and move
```
Comment (jpic) [inline docs/intro/reusable-apps.txt]: Are you sure there needs to be such file even on python 3 ?

File: docs/intro/reusable-apps.txt [hunk 2]
```diff
@@ -288,6 +288,70 @@ working. We'll now fix this by installing our new ``django-polls`` package.
...
+       }
+
+       TEMPLATES = [
+           {
+               'BACKEND': 'django.template.backends.django.DjangoTemplates',
+               'APP_DIRS': True,
+           },
+       ]
+
+       ROOT_URLCONF = 'tests.urls'
```
Comment (jpic) [inline docs/intro/reusable-apps.txt]: Maybe you can document the pytest alternative, create a conftest.py file as such:

```
from django.conf import settings    
    
    
def pytest_configure():    
    settings.configure(    
        INSTALLED_APPS=[    
            'django.contrib.contenttypes',    
            'django.contrib.auth',    
        ],    
        DATABASES=dict(default=dict(    
            ENGINE='django.db.backends.sqlite3',    
            NAME=':memory:',    
        ))    
    )    
```

**Comment by jpic** [comment]
Thanks for your contribution ! Interresting section to add to the docs, but I think it should be broader and shorter at the same time. For instance, you could use django-admin startproject and start a yourapp.test_project and then just set DJANGO_SETTINGS_MODULE=yourapp.test_project.test_project.settings to run tests, that would make the article much shorter.

At the same time it can be much broader by referering to djangopackages.com page about test tools, re-link about [settings override](https://docs.djangoproject.com/en/2.1/topics/testing/tools/#overriding-settings), show the pytest-django alternative, to be maybe more instructive.

**Comment by RaphaelKimmig** [comment]
> Thanks for your contribution ! Interresting section to add to the docs, but I think it should be broader and shorter at the same time. For instance, you could use django-admin startproject and start a yourapp.test_project and then just set DJANGO_SETTINGS_MODULE=yourapp.test_project.test_project.settings to run tests, that would make the article much shorter.

I'm not sure how that would work together with the rest of the article. 
In the basic intro we show users how to add tests to an app. 
So far the reusable app intro just throws those tests away and leaves the reusable app untested.
My goal here was to make sure that somebody following along would end up with a tested reusable app.

> At the same time it can be much broader by referering to djangopackages.com page about test tools, re-link about [settings override](https://docs.djangoproject.com/en/2.1/topics/testing/tools/#overriding-settings), show the pytest-django alternative, to be maybe more instructive.
I do think that talking about these things would be helpful. I am however not sure if this is the right place to do so.

**Comment by RaphaelKimmig** [comment]
I've rebased on current master and removed the reference to the `__init__.py`

File: docs/intro/reusable-apps.txt [hunk 3]
```diff
@@ -288,6 +288,70 @@ working. We'll now fix this by installing our new ``django-polls`` package.
...
     pip uninstall django-polls
 
+Testing your app
+================
+Since we moved the ``polls`` directory out of the project,
+we don't have the project's ``settings.py`` and ``urls.py`` anymore.
+We need another way to configure our app in order to test it.
+
+#. First we create a ``tests`` directory and move the tests over from the
+   ``polls`` directory. Because the tests now reside in a different package
```
Comment (carltongibson) [inline docs/intro/reusable-apps.txt]: `...different module...`?

File: docs/intro/reusable-apps.txt [hunk 4]
```diff
@@ -288,6 +288,70 @@ working. We'll now fix this by installing our new ``django-polls`` package.
...
+       urlpatterns = [
+           path('polls/', include('polls.urls')),
+       ]
+
+#. Now we can run the tests. ::
+
+    python -m django test --settings=tests.settings tests
+
+For more information on testing reusable applications see :doc:`Advanced testing topics
+<../topics/testing/advanced>`.
```
Comment (carltongibson) [inline docs/intro/reusable-apps.txt]: Would this be better linking to the `Using the Django test runner to test reusable applications` section directly?

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**